source/connection/i_mplb.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "flow/i_flow.hpp"
+#include "packet.hpp"
+#include "types.hpp"
+
+namespace sim {
+
+// Captures per-packet flow-level statistics used by MPLB logic
+struct FlowSample {
+    TimeNs ack_recv_time;
+    TimeNs packet_sent_time;
+    std::uint32_t packets_in_flight;
+    double curr_cwnd;
+    SpeedGbps delivery_rate;
+    // add other metrics here as needed (e.g., RTT variance, ECN, etc.)
+};
+
+class IMPLB {
+public:
+    virtual void add_flow(std::shared_ptr<IFlow> flow) = 0;
+    virtual void remove_flow(const std::shared_ptr<IFlow>& flow) = 0;
+    virtual void notify_ack_received(const std::shared_ptr<IFlow>& flow, FlowSample metrics) = 0;
```

Comment:
Naming suggestion (some flows might not use acks; strange but why not):
```suggestion
    virtual void notify_packet_confirmed(const std::shared_ptr<IFlow>& flow, FlowSample metrics) = 0;
```